test/langtools/tools/doclint/CrashInAnnotateTest.out

```diff
@@ -0,0 +1,9 @@
+CrashInAnnotateTest.java:10:5: compiler.err.proc.messager: annotations not allowed
+CrashInAnnotateTest.java:14:5: compiler.err.proc.messager: annotations not allowed
+CrashInAnnotateTest.java:16:10: compiler.warn.proc.messager: no comment
+CrashInAnnotateTest.java:19:5: compiler.err.proc.messager: syntax error in reference
```

Comment:
I think that this errors are because the annotation is fully qualified

Reply:
Yes and no. The parts other than the arguments are parsed in a different way that doesn't allow a space, see DocCommentParser.java:417 . Since there can be no space it is impossible to insert a type annotation in a type. I have updated the test with a few more examples.

I'm not looking to unify parsing in this PR, IMO it is enough to 1) not crash and 2) provide a friendly message.